src/streamlink/plugins/douyutv.py

```diff
@@ -121,56 +131,15 @@ def _get_streams(self):
             self.logger.info("Stream currently unavailable.")
             return
 
-        ts = int(time.time() / 60)
-        did = uuid.uuid4().hex.upper()
-
-        #use new API key and modified MD5 algorithm
-        sign = stupidMD5(("{0}{1}{2}{3}".format(channel, did, LAPI_SECRET, ts)))
-
-        data = {
-            "cdn": "ws",
-            "rate": "0",
-            "tt": ts,
-            "did": did,
-            "sign": sign
-        }
-
-        res = http.post(LAPI_URL.format(channel), data=data)
-        room = http.json(res, schema=_lapi_schema)
-
-        url = "{room[rtmp_url]}/{room[rtmp_live]}".format(room=room)
-        stream = HTTPStream(self.session, url)
-        yield "source", stream
-
-        data = {
-            "cdn": "ws",
-            "rate": "2",
-            "tt": ts,
-            "did": did,
-            "sign": sign
-        }
-
-        res = http.post(LAPI_URL.format(channel), data=data)
-        room = http.json(res, schema=_lapi_schema)
-
-        url = "{room[rtmp_url]}/{room[rtmp_live]}".format(room=room)
-        stream = HTTPStream(self.session, url)
-        yield "middle", stream
-
-        data = {
-            "cdn": "ws",
-            "rate": "1",
-            "tt": ts,
-            "did": did,
-            "sign": sign
-        }
-
-        res = http.post(LAPI_URL.format(channel), data=data)
-        room = http.json(res, schema=_lapi_schema)
-
-        url = "{room[rtmp_url]}/{room[rtmp_live]}".format(room=room)
-        stream = HTTPStream(self.session, url)
-        yield "low", stream
+        room_source = self._get_room_json(channel, 0)
+        yield "source", HTTPStream(self.session, room_source['live_url'])
+        yield "source", HLSStream(self.session, room_source['hls_url'])
 
+        room_middle = self._get_room_json(channel, 2)
+        yield "middle", HTTPStream(self.session, room_middle['live_url'])
+        yield "middle", HLSStream(self.session, room_middle['hls_url'])
```

Comment:
since you changed it to `medium`
you missed some old `middle` stuff here and

`room_middle`

`# 0:source 2:middle 1:low`